web/src/message_util.ts

```diff
@@ -16,23 +19,33 @@ export function do_unread_count_updates(messages, expect_no_new_unreads = false)
     }
 }
 
-export function add_messages(messages, msg_list, opts) {
+export function add_messages(
+    messages: RawMessage[],
```

Comment:
I don't think these are `RawMessage` objects; 

At least, they've all been processed via `message_helper.process_new_message` ... which probably would be a good one to convert to `TypeScript` and typed as converting `RawMessage` into `Message` objects? I've not double-checked but it seems like that might be what it does.

Reply:
It would be helpful if we could have this conversation in #28373 instead, which I linked to in the PR description

Reply:
moved conversation to [here](https://github.com/zulip/zulip/pull/28373#discussion_r1437868361)